test/unit/resources/aws_ebs_volume_test.rb

```diff
@@ -0,0 +1,75 @@
+require 'helper'
+
+class TestEbs < Minitest::Test
+  Id = 'volume-id'.freeze
+
+  def setup
+    @mock_conn = Minitest::Mock.new
+    @mock_client = Minitest::Mock.new
+    @mock_resource = Minitest::Mock.new
+    @mock_iam_resource = Minitest::Mock.new
+
+    @mock_conn.expect :ec2_client, @mock_client
+    @mock_conn.expect :ec2_resource, @mock_resource
+    @mock_conn.expect :iam_resource, @mock_iam_resource
+  end
+
+  def test_that_id_returns_id_directly_when_constructed_with_an_id
+    assert_equal Id, AwsEbsVolume.new(Id, @mock_conn).id
+  end
+
+  def test_that_id_returns_fetched_id_when_constructed_with_a_name
+    mock_volume = Minitest::Mock.new
+    mock_volume.expect :nil?, false
```

Comment:
It seems you're mixing expectations and assertions here. Which is fine, but you could also do:

```
refute(mock_volume.nil?)
```

We try to use either assertions or expectations throughout in a file. We are working to make our preference more clear in the style guide, so this isn't your fault.

Would you care to convert your tests to one or the other?

A list of the assertions available: http://ruby-doc.org/stdlib-2.0.0/libdoc/minitest/rdoc/MiniTest/Assertions.html